c10/core/TensorImpl.cpp

```diff
@@ -85,6 +85,40 @@ bool TensorImpl::compute_contiguous() const {
   return is_contiguous;
 }
 
+bool TensorImpl::compute_channels_last_contiguous() const {
+  if (dim() == 4) {
+    int64_t expected = 1;
+    for (auto& d : {1, 3, 2, 0}) {
+      if (size(d) != 1) {
+        if (stride(d) == expected) {
+          expected *= size(d);
+        } else {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+  return false;
+}
+
+bool TensorImpl::compute_strides_like_channels_last() const {
```

Comment:
are you trying to cover slicing of channel-last tensors? does it actually occur in practice?

Reply:
Slicing and bounding. Both might happen in pre-processing.